gocourse02/zoo/snake.go

```diff
@@ -0,0 +1,28 @@
+package zoo
+
+import (
+	"fmt"
+	"reflect"
+)
+
+type Snake struct {
+	*Animal
+	Length int
+}
+
+func NewSnake(animal *Animal, length int) *Snake {
+	return &Snake{
+		Animal: animal,
+		Length: length,
+	}
+}
+
+func (t *Snake) Reproduce(otherSnake *Snake) *Snake {
```

Comment:
Для консистенції:
```suggestion
func (s *Snake) Reproduce(snake *Snake) *Snake {
```

https://go.dev/wiki/CodeReviewComments#receiver-names

`s` від `snake`, `otherSnake` - занадто довга назва, можна `snake`